extension/test/test-logger.js

```diff
@@ -27,7 +33,12 @@ var verifyAsyncOutput = function (test, logger, testFunc) {
     }
     finally {
       istream.close();
-      message = JSON.parse(message);
+      try {
+        message = JSON.parse(message);
+      } catch (e) {
+        Cu.reportError("Invalid JSON string: " + message);
+        throw e;
+      }
```

Comment:
There is `assert.throws()` which we might be able to use here. But not sure if that method is logging failures to the Browser console. Maybe worth a try.